llama_stack/cli/stack/build.py

```diff
@@ -49,7 +49,7 @@ def _add_arguments(self):
             type=str,
             help="Image Type to use for the build. If not specified, will use the image type from the template config.",
             choices=[e.value for e in ImageType],
-            default=ImageType.CONDA.value,
+            default=None,  # no default so we can detect if a user specified --image-type and override image_type in the config
```

Comment:
What if not specified? Is that path correctly exercise in the CI or did you test it?

Reply:
if the default is set to conda then we can't later tell if a user specified --image-type or we're seeing the default. this caused an earlier version's test case failure, it resulted in the image_type always being set to conda.

image type precedence -
 - --image-type on command line
 - image_type in config
 - conda (via BuildConfig's image_type default, see https://github.com/meta-llama/llama-stack/blob/main/llama_stack/distribution/datatypes.py#L321)